### PR TITLE
Request validation for cron

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Towards #4284

### CODE REVIEW

File: server/routes/api/cron/schema.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    ctx.addIssue({
+      code: z.ZodIssueCode.custom,
+      message: "token is required",
+      fatal: true,
+    });
+
+    return z.NEVER;
+  }
+
+  const token = (req.body.token ?? req.query.token) as string;
```
Comment (tommoor) [inline server/routes/api/cron/schema.ts:27]: Interesting, not sure how I feel about having the authentication inside data validation, feels like this is mixing concerns?

File: server/routes/api/cron/cron.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    success: true,
+  };
+};
+
+router.get("cron.:period", validate(T.CronSchema), cronHandler);
+router.post("cron.:period", validate(T.CronSchema), cronHandler);
+
+// For backwards compatibility
+router.get("utils.gc", cronHandler);
+router.post("utils.gc", cronHandler);
```
Comment (tommoor) [inline server/routes/api/cron/cron.ts:29]: Missing validate middleware on these

File: server/routes/api/cron/schema.ts (hunk 1 above)
Comment (apoorv-mishra) [reply server/routes/api/cron/schema.ts:27]: You sound quite right. Perhaps we can retain `token` presence and `token.length` checks here and move the `crypto` checks inside the API handler?

Comment (apoorv-mishra) [reply server/routes/api/cron/schema.ts]: Separated concerns! Kept `token.length` check _with_ `crypto` check, made sense _semantically_ :+1:

**Review by tommoor — APPROVED** [review]

**Pull request merged**